src/pieces/utils.py

```diff
@@ -0,0 +1,50 @@
+import json
+from bs4 import BeautifulSoup
+import os
+import re
+
+from pieces.settings import Settings
+
+# Used to create a valid file name when opening to "Opened Snippets"
+def sanitize_filename(name):
```

Comment:
this strategy is probably okay however a better one might be to use url encoding for this.

For example, instead of encoding ' ' as '_' we can encode it as '%20'

Reply:
This is the name of the file!